src/erlcloud_mms.erl

```diff
@@ -0,0 +1,257 @@
+
+-spec batch_meter_usage(ProductCode :: binary(),
+                        UsageRecords :: proplist()) -> proplist().
+batch_meter_usage(ProductCode, UsageRecords) ->
+    batch_meter_usage(ProductCode, UsageRecords, default_config()).
+
+-spec batch_meter_usage(ProductCode :: binary(),
+                        UsageRecords :: proplist(),
+                        aws_config()) -> proplist().
+batch_meter_usage(ProductCode, UsageRecords, Config) ->
+    Json = [{<<"ProductCode">>, ProductCode}, {<<"UsageRecords">>, UsageRecords}],
+
+    request(Config,
+            "AWSMPMeteringService.BatchMeterUsage",
+            Json).
+%%%------------------------------------------------------------------------------
+%%% Meter Usage
+%%%------------------------------------------------------------------------------
+
+-spec meter_usage(ProductCode :: binary(),
+                  Timestamp :: number(),
+                  UsageDimension  :: binary(),
+                  UsageQuantity :: number(),
+                  DryRun  :: boolean()) -> proplist().
+meter_usage(ProductCode, Timestamp, UsageDimension, UsageQuantity, DryRun) ->
+    meter_usage(ProductCode, Timestamp, UsageDimension, UsageQuantity, DryRun, default_config()).
+
+-spec meter_usage(ProductCode :: binary(),
+                  Timestamp :: number(),
+                  UsageDimension  :: binary(),
+                  UsageQuantity :: number(),
+                  DryRun  :: boolean(),
+                  aws_config()) -> proplist().
+meter_usage(ProductCode, Timestamp, UsageDimension, UsageQuantity, DryRun, Config) ->
+    Json = [{<<"DryRun">>, DryRun},
+            {<<"ProductCode">>, ProductCode},
+            {<<"Timestamp">>, Timestamp},
+            {<<"UsageDimension">>, UsageDimension},
+            {<<"UsageQuantity">>, UsageQuantity}],
+
+    request(Config,
+            "AWSMPMeteringService.MeterUsage",
+            Json).
+
+%%%------------------------------------------------------------------------------
+%%% Resolve Customer
+%%%------------------------------------------------------------------------------
+
+-spec resolve_customer(RegistrationToken :: binary()) -> proplist().
+resolve_customer(RegistrationToken) ->
+    resolve_customer(RegistrationToken, default_config()).
+
+-spec resolve_customer(RegistrationToken :: binary(),
+                       aws_config()) -> proplist().
+resolve_customer(RegistrationToken, Config) ->
+    Json = [{<<"RegistrationToken">>, RegistrationToken}],
+
+    request(Config,
+            "AWSMPMeteringService.ResolveCustomer",
+            Json).
+
+%%%------------------------------------------------------------------------------
+%%% Request
+%%%------------------------------------------------------------------------------
+
+-type operation() :: string().
+-type json_return() :: {ok, json_term()} | {error, term()}.
+
+-spec request(aws_config(), operation(), json_term()) -> json_return().
+request(Config, Operation, Json) ->
```

Comment:
what is the point of this wrapper given nobody uses `request2` directly?
please simplify